AnnService/inc/Core/Common/WorkSpace.h

```diff
@@ -135,54 +135,71 @@ namespace SPTAG
         };
 
         class DistPriorityQueue {
-            std::unique_ptr<float[]> m_data;
             int m_size;
+            std::unique_ptr<float[]> m_data;
+            int m_length;
             int m_count;
+            
         public:
             DistPriorityQueue() {}
 
             void Resize(int size_) {
-                m_data.reset(new float[size_ + 1]);
-                for (int i = 0; i <= size_; i++) m_data[i] = MaxDist;
                 m_size = size_;
+                m_data.reset(new float[size_ + 1]);
+                
+                m_data[1] = MaxDist;
```

Comment:
what's the usage of m_data{0], why don't you use m_data[0] to store MaxDist?

Reply:
to make the tree traverse faster, the root index is better to be 1.

Reply:
then m_data[0] is unused?

Reply:
yes

Reply:
could you consider to use clean m_data to reuse original array and avoid extra heap allocation?